src/libcore/sync/atomic.rs

```diff
@@ -2518,6 +2546,7 @@ pub fn fence(order: Ordering) {
 #[inline]
 #[stable(feature = "compiler_fences", since = "1.21.0")]
 pub fn compiler_fence(order: Ordering) {
+    // SAFETY: doesn't compile to machine code
```

Comment:
That's not an argument. Safety is defined on the Rust Abstract Machine, and you [cannot use machine-code-level reasoning to argue about UB](https://www.ralfj.de/blog/2019/07/14/uninit.html).

Reply:
Great article!
I'm a still a little unsure what to put here. Are we OK just assuming these intrinisics are safe to use anywhere?

Reply:
The intrinsic could be safe, it just isn't because we didn't bother marking it as such. So the comment should be the same as for the atomic fence, I think.